daq/faucet_events.py

```diff
@@ -0,0 +1,53 @@
+from collections import namedtuple
+
+EVENT_TYPE_PORT_STATE = "Port State"
+EVENT_TYPE_PORT_LEARN = "Port Learn"
+EVENT_TYPE_CONFIG_CHANGE = "Config Change"
+
+all_events = {}
+
+
+def save_all_events(f):
+    def wrapped(*args, **kwargs):
+        res = f(*args, **kwargs)
+        print(all_events)
+        return res
+    return wrapped
+
+
+@save_all_events
+def process_port_state(event):
+    all_events\
+        .setdefault(event.dpid, {})\
+        .setdefault(EVENT_TYPE_PORT_STATE, {})\
+        .setdefault(event.port, [])\
+        .append({"ts":event.timestamp, "acitve": event.active})
+
+
+@save_all_events
+def process_port_learn(event):
+    all_events\
+        .setdefault(event.dpid, {})\
+        .setdefault(EVENT_TYPE_PORT_LEARN, {})\
+        .setdefault(event.port, [])\
+        .append({"ts":event.timestamp, "mac":event.mac})
```

Comment:
is this collecting all events in a list?  If so, not the right behavior -- it should be just representing the last state, and then keeping a count of the number of updates.